nixos/modules/services/databases/influxdb2.nix

```diff
@@ -0,0 +1,50 @@
+{ config, lib, pkgs, ... }:
+with lib;
+let
+  cfg = config.services.influxdb2;
+  configFile = pkgs.writeText "config.json" (builtins.toJSON cfg.config);
+in
+{
+  options = {
+    services.influxdb2 = {
+      enable = mkEnableOption "Whether to enable the influxdb2 server";
+      package = mkOption {
+        default = pkgs.influxdb2;
+        defaultText = "pkgs.influxdb2";
+        description = "influxdb2 derivation to use";
+        type = types.package;
+      };
+      config = mkOption {
```

Comment:
I think this should be called "settings" to be consistent with https://github.com/NixOS/rfcs/blob/master/rfcs/0042-config-option.md